### PR TITLE
[Request for feedback] Heroku fix

### PR DESCRIPTION
**Pull request by davisshaver**
## What does this PR do?
The Deploy to Heroku button had been removed following a bug report https://github.com/coralproject/talk/issues/1685.

This PR makes a few tweaks to the Talk repo to facilitate one-click deployment with Heroku.

- Omits `package-lock.json` on Heroku to use Yarn instead.

- Silences rewrite env errors, which throws for the Mailgun SMTP password even if one is set through Heroku. Maybe this is due to when the Heroku env is available.

## How do I test this PR?
- Create a new app on Heroku using [this link](https://dashboard.heroku.com/new?template=https%3A%2F%2Fgithub.com%2Fdavisshaver%2Ftalk-heroku&env[TALK_FACEBOOK_APP_ID]=ignore&env[TALK_FACEBOOK_APP_SECRET]=ignore&env[REWRITE_ENV_SILENT]=true).
- The app should build without error.

### CODE REVIEW

**Comment by davisshaver** [comment]
Netfily CI depends on `yarn.lock` being present to make `yarn` available to the environment. https://www.netlify.com/docs/build-settings/

**Comment by davisshaver** [comment]
Not sure why Netfily or the integration tests failed.

**Comment by kgardnr** [comment]
@davisshaver Thanks for this!! The netlify build log is complaining about yarn install. Can you try running yarn install locally and see if you get an error?

**Comment by davisshaver** [comment]
@kgardnr Netifly is working now, although the integration test is still failing. I don't believe it's related to the changeset though.

File: package.json [hunk 1]
```diff
@@ -63,6 +63,7 @@
     "apollo-server-express": "^1.2.0",
     "apollo-utilities": "^1.0.3",
     "app-module-path": "^2.2.0",
+    "archiver": "^2.1.1",
```
Comment (wyattjoh) [inline package.json]: These dependencies are resolved during the build process, and should not be included in the base `package.json`.

**Review by wyattjoh — CHANGES_REQUESTED** [review]
The `pakage-lock.json` should not be committed to this repository as we're using yarn in this configuration.

As such, it shouldn't be required to add the `.slugignore` file either.

**Comment by davisshaver** [comment]
🤦‍♂️ Thanks @wyattjoh! I'll get this cleaned up.

~~regarding the packages, do you have any advice on how to debug why they might not be resolved during the build process?~~

**Comment by davisshaver** [comment]
Cleaned this up and with Yarn working properly, the compile process work as expected. Build log (with failing example afterwards demonstrating the ENV issue): https://gist.github.com/davisshaver/d5bbf6c404bfaaa39ab76a1acf5db4b4

**Comment by wyattjoh** [comment]
Thanks for your patience @davisshaver! As a more permanent solution (that should also be Heroku independent) could you revert the `app.json` and instead replace it with the following?

```diff
﻿﻿diff --git a/services/env.js b/services/env.js
index b40a44a83..b3aaa42b3 100644
--- a/services/env.js
+++ b/services/env.js
@@ -1,3 +1,7 @@
+// Ensure that we always process the rewrites silently, otherwise it may fail in
+// environments like Heroku.
+process.env.REWRITE_ENV_SILENT = 'TRUE';
+
 // Perform rewrites to the runtime environment variables based on the contents
 // of the process.env.REWRITE_ENV if it exists. This is done here as it is the
 // entrypoint for the entire applications configuration.

```

**Comment by davisshaver** [comment]
@wyattjoh good idea, all set for another look!

**Review by wyattjoh — APPROVED** [review]

**Comment by wyattjoh** [comment]
Looks great! Thanks!

**Pull request merged**